pennylane/ops/qubit/state_preparation.py

```diff
@@ -345,6 +352,14 @@ def _unflatten(cls, data, metadata):
         return cls(*data, **dict(metadata[0]), wires=metadata[1])
 
     def state_vector(self, wire_order: Optional[WiresLike] = None):
+
+        if self.is_sparse:
+            return (
+                _sparse_statevec_permute(self.parameters[0], self.wires, wire_order)
+                if wire_order
+                else self.parameters[0]
```

Comment:
Is this the first term of a list with all initialization parameters?
If yes, what happens if someone provides `qml.StatePrep(wires=[1, 2], state=init_state)` instead of  `qml.StatePrep(init_state, wires=[1, 2])`?

Reply:
Could we return `self.state` in this case?

Reply:
@AmintorDusko Given that Python has been able to allow user to appoint positional args in a keyword way I don't think this will cause any problem atl. Directly running the two different instantiations also gave me the same `parameters[0]` to be the input `init_state` (kick me awake if this went wrong somehow)

But thanks for bringing it up and  this indeed reminds me that probably we also need to check some methods supported in parents classes that makes use of the `parameters[0]`; highly likely there would be something like overloaded `__equal__` or similar that couldn't be gauranteed any more.

Reply:
@JerryChen97, good to know! Thank you for clarifying!